core/circuitbreaker/stat_slot.go

```diff
@@ -0,0 +1,29 @@
+package circuitbreaker
+
+import (
+	"github.com/alibaba/sentinel-golang/core/base"
+)
+
+// MetricStatSlot add statistic metric for circuit breaker
+// statistic is based on completed.
+type MetricStatSlot struct {
+}
+
+func (c *MetricStatSlot) OnEntryPassed(_ *base.EntryContext) {
+	// Do nothing
+	return
+}
+
+func (c *MetricStatSlot) OnEntryBlocked(_ *base.EntryContext, _ *base.BlockError) {
+	// Do nothing
+	return
+}
+
+func (c *MetricStatSlot) OnCompleted(ctx *base.EntryContext) {
+	res := ctx.Resource.Name()
```

Comment:
We need to check whether this request has been blocked. If blocked, the request should not be recorded to the circuit breakers.